core-bundle/src/Migration/Version412/AllowedExcludedFieldsMigration.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration\Version412;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class AllowedExcludedFieldsMigration extends AbstractMigration
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function shouldRun(): bool
+    {
+        $schemaManager = $this->connection->getSchemaManager();
+
+        if (!$schemaManager->tablesExist('tl_user_group')) {
+            return false;
+        }
```

Comment:
```suggestion
        if (!$schemaManager->tablesExist('tl_user_group')) {
            return false;
        }
        
        $columns = $schemaManager->listTableColumns('tl_user_group');

        if (!isset($columns['id'], $columns['modules'], $columns['filemounts'], $columns['fop'], $columns['alexf'])) {
            return false;
        }
```

Migrations should be as robust as possible IMO. Like if someone has one of the fields unset in the DCA, the migration should not run into an error.

Reply:
I guess we do not have to check for `$columns['id']`, because Contao will not work without this column. But otherwise agreed. 👍🏻

Reply:
Added in 9730741387614720e5e628f53c96dd94457db637.